tests/end2end_test/g_spreadsheet_end_to_end_test.py

```diff
@@ -38,14 +38,19 @@ def test_dag_runs_data_imported():
 
     AIRFLW_API.unpause_dag(TARGET_DAG_ID)
     execution_date = AIRFLW_API.trigger_dag(dag_id=DAG_ID)
-    dag_status: str
-    while True:
-        dag_status = AIRFLW_API.get_dag_status(DAG_ID, execution_date)
-        if dag_status not in {"running", "queued"}:
-            break
+    is_running = True
+    while is_running:
+        is_running = AIRFLW_API.is_dag_running(DAG_ID, execution_date)
+        if not is_running:
+            is_running = AIRFLW_API.is_triggered_dag_running(
```

Comment:
I believe this section is slightly different. It is because the controller DAG is triggering sub DAGs.
Ideally we would put the loop into a function (e.g. `wait_for_dag_run_success` or similar).
Then we could call it with the controller DAG id, and then then child DAG id.